package/kernel/linux/modules/netsupport.mk

```diff
@@ -716,15 +716,16 @@ $(eval $(call KernelPackage,mppe))
 
 
 SCHED_MODULES = $(patsubst $(LINUX_DIR)/net/sched/%.ko,%,$(wildcard $(LINUX_DIR)/net/sched/*.ko))
-SCHED_MODULES_CORE = sch_ingress sch_fq_codel sch_hfsc sch_htb sch_tbf cls_fw cls_route cls_flow cls_tcindex cls_u32 em_u32 act_mirred act_skbedit
-SCHED_MODULES_FILTER = $(SCHED_MODULES_CORE) act_connmark sch_netem
+SCHED_MODULES_CORE = sch_ingress sch_fq_codel sch_hfsc sch_htb sch_tbf cls_basic cls_fw cls_route cls_flow cls_tcindex cls_u32 em_u32 act_mirred act_skbedit cls_matchall
+SCHED_MODULES_FILTER = $(SCHED_MODULES_CORE) act_connmark sch_netem em_ipset cls_bpf act_bpf
 SCHED_MODULES_EXTRA = $(filter-out $(SCHED_MODULES_FILTER),$(SCHED_MODULES))
 SCHED_FILES = $(patsubst %,$(LINUX_DIR)/net/sched/%.ko,$(filter $(SCHED_MODULES_CORE),$(SCHED_MODULES)))
 SCHED_FILES_EXTRA = $(patsubst %,$(LINUX_DIR)/net/sched/%.ko,$(SCHED_MODULES_EXTRA))
 
 define KernelPackage/sched-core
   SUBMENU:=$(NETWORK_SUPPORT_MENU)
   TITLE:=Traffic schedulers
+  DEPENDS:=@!LINUX_3_18 @!LINUX_4_1 @!LINUX_4_4
```

Comment:
Hmm. I was being cautious to prevent `cls_matchall` usage prior to kernel-4.9. Also, I've been testing on both `openwrt-18.06` and `master`, and wanted to allow for a simple backport.

But I see you're right. After double-checking, it seems using only `@!LINUX_3_18` should work for both `openwrt-18.06` and `master` branches, so I'll drop the `@!LINUX_4_1 @!LINUX_4_4` portion.

Reply:
I also realized that since the iproute2 `tc` utility is dependent on `kmod-sched-core`, it won't be build-able for linux-3.18 targets (where `kmod-sched-core` is disabled).

@hauke  Is the above OK given all the `master` targets using 3.18 are marked "source-only" or "broken"?

Reply:
Add kernel 4.14 and 4.19

Reply:
These kernels are already supported, so please clarify your comment further.